File: src/main/java/net/hectus/neobb/turn/default_game/flower/TWhiteTulip.java
package net.hectus.neobb.turn.default_game.flower;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.ColdClazz;
import org.bukkit.block.Block;

import java.util.List;

public class TWhiteTulip extends FlowerTurn {
    public TWhiteTulip(NeoPlayer player) { super(player); }
    public TWhiteTulip(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 4;
    }

    @Override
    public void apply() {
        player.game.allowedClazzes().add(ColdClazz.class);
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.ExtraTurn());
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/flower/TWhiteTulip.kt
package net.hectus.neobb.modes.turn.default_game.flower

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.ColdClazz
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class TWhiteTulip(data: Block?, cord: Cord?, player: NeoPlayer?) : FlowerTurn(data, cord, player) {
    override val cost: Int = 4

    override fun apply() {
        player!!.game.allowed.add(ColdClazz::class)
    }

    override fun buffs(): List<Buff<*>> {
        return listOf(ExtraTurn())
    }
}
